polaris-react/src/components/IndexFilters/hooks/useSetIndexFiltersMode/useSetIndexFiltersMode.tsx

```diff
@@ -1,11 +1,19 @@
-import {useState} from 'react';
+import {useEffect} from 'react';
 
-import {IndexFiltersMode} from '../../types';
+import {
+  useIndexFiltersManager,
+  IndexFiltersMode,
+} from '../../../../utilities/index-filters';
 
 export function useSetIndexFiltersMode(
   defaultMode: IndexFiltersMode = IndexFiltersMode.Default,
 ) {
-  const [mode, setMode] = useState<IndexFiltersMode>(defaultMode);
+  const {mode, setMode} = useIndexFiltersManager();
+
+  useEffect(() => {
+    setMode(defaultMode);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [defaultMode]);
```

Comment:
Why do we need both a `useSetIndexFiltersMode` and `useIndexFiltersManager`? With the solution being proposed could we not just use `useIndexFiltersManager`?

Additionally, is this `useEffect` necessary? It seems it's changing the behavior and meaning of `defaultMode` by allowing `defaultMode` to reset the state on change. I'm inclined to think it should not be doing that and should continue to be utilized as a default value, then utilize `setMode` to updates by consumers.

Reply:
@mattkubej We don't want to remove `useSetIndexFiltersMode` as this would become a breaking change to Polaris. However, I feel we could rename the `useIndexFiltersManager` to `useSetIndexFiltersMode` and then remove the current `useSetIndexFiltersMode` from within the `IndexFilters` component to remove this duplication.

As for the useEffect, we do need this as we want the `useSetIndexFiltersMode` to be able to update the context when the hook initialized. I've changed this so it only runs on the mounting of the hook by removing the dependency array so it doesn't update the context on state change.